### ISSUE TITLE
Unexpected Output Using `slam_scaled` Model with `eval.py`

### ISSUE THREAD

**Issue by ankit-chotaliya** [slp-rl/slamkit#5]
#### **Description**  
I am experiencing unexpected output when running the `slam_scaled` model using the evaluation script. The generated output does not match the expected result, and instead, I get noise at the beginning of the output file instead of a resynthesized version of the input.  

#### **Steps that I followed**  
1. **Input and output Audio:**  
   - [Input and output wav link](https://drive.google.com/drive/folders/1zEj79PDm4fRNnK7usZUOwkjkDHpi0Ksg?usp=sharing) 
   - I have ran same command for two times with same audio sample.
2. **Command Used:**  
   ```bash
   python3.12 cli/eval.py tokeniser=unit_hubert_25 metric=generate batch_size=32 model.pretrained_model=slprl/slam_scaled metric.data_path=/home/ubuntu/slamkit/samples/sample_1.wav vocoder=vocoder_hubert_25
   ```
3. **Environment Details:**  
   - **GPU:** [AWS g5.4xlarge](https://instances.vantage.sh/aws/ec2/g5.4xlarge)  
   - **Python Version:** 3.12  
   - **Installed Dependencies:**  
     ```bash
     python3.12 -m pip list | grep -E 'torch|torchaudio|torchvision|transformers|datasets|omegaconf|hydra-core|accelerate|numpy|ninja|wandb|joblib|trl|scikit-learn'
     ```
     ```
     accelerate               1.4.0  
     datasets                 3.3.2  
     hydra-core               1.3.2  
     joblib                   1.4.2  
     ninja                    1.11.1.3  
     numpy                    2.2.3  
     omegaconf                2.3.0  
     scikit-learn             1.6.1  
     torch                    2.6.0  
     torchaudio               2.6.0  
     torchvision              0.21.0  
     transformers             4.49.0  
     trl                      0.15.2  
     wandb                    0.19.7  
     ```
4. **Code Commit Used:**  
   - **Commit Hash:** https://github.com/slp-rl/slamkit/tree/12fb0c6890f6fc61b88144f8814822932416209c
   - **Any Modifications to Config:** None  

#### **Issue Observed**  
- The expected behavior is that the beginning of the output file should be a resynthesized version of the input.  
- However, the generated output contains noise instead.  

#### **Additional Information**  
- No modifications were made to the config files.  
- Let me know if any other details are needed.

**Comment by gallilmaimon** [comment]
Hello,
We appreciate your interest in this repository :)

I was unable to reproduce your unexpected behaviour. I am adding here the output I got when using your provided audio prompt:
[generate_0.wav.zip](https://github.com/user-attachments/files/19143533/generate_0.wav.zip)

I did need to change your command slightly as trying to generate 5 files (the default number in `metric.num_files` for `generate.yaml`) with a path to a single wav file (e.g. `metric.data_path=/home/ubuntu/slamkit/samples/sample_1.wav`) raises a stop iteration error. For this reason I am surprised that the command you provided ran without errors. Did this require any changes to the config file or to the scripts? Or is this a clean clone of the repository?

While I don't have access to the exact same GPU that you used I don't think that this is GPU specific, as even when running on CPU (by adding `device=cpu` to the command) I got decent results:
[generate_0_2.wav.zip](https://github.com/user-attachments/files/19143585/generate_0_2.wav.zip)

My biggest suspicion is something in the backend that loads the audio, or the HuBERT model loading. While I try to give some minimal tests to check this I would appreciate if you could try with a clean clone to the repo (with no modifications to the scripts or configs) if this not already the case.

```
python cli/eval.py tokeniser=unit_hubert_25 metric=generate batch_size=32 model.pretrained_model=slprl/slam_scaled metric.data_path=/home/ubuntu/slamkit/samples/sample_1.wav vocoder=vocoder_hubert_25 metric.num_files=1
```

**Comment by ankit-chotaliya** [comment]
Thank you for looking into this.  

I have re-run the code after cleaning all model caches and ensuring a clean clone of the repository (with no modifications to scripts or configs). However, I am still encountering the same unexpected output.  

Here is the new output:  

[generate_0.wav.zip](https://github.com/user-attachments/files/19157085/generate_0.wav.zip)
 

Let me know if you have any further suggestions or if I can provide more details to help debug this.

I have ran following command to generate output (which is exactly same that you have provided)

`python3.12 cli/eval.py tokeniser=unit_hubert_25 metric=generate batch_size=32 model.pretrained_model=slprl/slam_scaled metric.data_path=/home/ubuntu/slamkit/samples/sample_1.wav vocoder=vocoder_hubert_25 metric.num_files=1`

**Comment by gallilmaimon** [comment]
Thanks for the provided info :) My main two suspicions would be something about audio loading or something about the HuBERT encoder.

In order to try debug the audio loading, could you do the following:
- Specify the Operating System that you are using + which backend torchaudio is using:
```py
import torchaudio
print(torchaudio.get_audio_backend())
```
- Specify the shape and sample rate of the loaded audio file, either add print after these lines or load it independently with the same torchaudio.load command:
https://github.com/slp-rl/slamkit/blob/12fb0c6890f6fc61b88144f8814822932416209c/slamkit/metric/generative_metric.py#L33

- Try loading and then saving the audio in order to make sure that it seems identical:
```py
import torchaudio
input_path = "example.wav"  # Replace with your actual file
output_path = "output.wav"

waveform, sample_rate = torchaudio.load(input_path)
torchaudio.save(output_path, waveform, sample_rate)
```

Hope one of these would be the issue. However if it is in fact the HuBERT encoding, one way to make sure would be to encode the example audio and compare to the provided expected output. Specifically, run:
```
python cli/extract_features.py data_path=example_data/audio ext=flac out_path=example_data/features2.jsonl batch_size=16 tokeniser=unit_hubert_25 tokeniser.feature_extractor.compile=true num_workers=4
```
And then compare the output at `example_data/features2.jsonl` with [example_data/features.jsonl](https://github.com/slp-rl/slamkit/blob/main/example_data/features.jsonl). While this is not necessarily identical, it should be very very similar.

One more question - how was the wav file you provided generated? The samples in the demo page were mp3 versions of the prompts I used (compressed so the site would load them quickly), and while the code works properly on my side with your file it could maybe help explain why it isn't loaded properly (as some backends of torchaudio have known issues with some versions of audio files).

**Comment by ankit-chotaliya** [comment]
hey this issue has been resolved.

thank you for your help. 

I have re ran the clean code as you mentioned and it ran well in macOS(sequoia 15.3.1).
